### PR TITLE
do not create initial entity

### PR DESCRIPTION
**Pull request by AlvaroVega**
Initial entity was created when a new device is created or just a measure for a new unprovisioned device comes.
In both cases  a new entity (with some  attrs with null values) is created.

Needs https://github.com/telefonicaid/iotagent-node-lib/pull/1111

- [x] Do not create initial entity when appendMode is true (by default) and not NGSILD
- [x] Upgrade all tests (/test/unit/ngsiv2/*)
- [x] Analyze bidirectional and commands case: will not create initial entity
- [x] align with master after merge https://github.com/telefonicaid/iotagent-node-lib/pull/1111
- [x] Upgrade agent tests (related agent PRs will be needed)

### CODE REVIEW

**Comment by AlvaroVega** [comment]
Maybe prevent to create initial entity is only possible when appendMode is true, otherwise old behaviour should be keept (or appendMode false will not work)

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -1,3 +1,4 @@
+- Add: Do not create initial entity when a new device is provisioned (just when appendMode is enabled)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE:1]:
```suggestion
- Add: do not create initial entity when a new device is provisioned and appendMode is false
```

(If I have understood the functionality correctly...)

Btw, no issue number associated?

File: test/unit/ngsiv2/provisioning/device-provisioning-api_test.js [hunk 2]
```diff
@@ -336,14 +336,14 @@ describe('NGSI-v2 - Device provisioning API: Provision devices', function () {
...
-                )
-                .reply(204);
+            // contextBrokerMock = nock('http://192.168.1.1:1026')
+            //     .matchHeader('fiware-service', 'smartgondor')
+            //     .matchHeader('fiware-servicepath', '/gardens')
+            //     .post(
+            //         '/v2/entities?options=upsert',
+            //         utils.readExampleFile('./test/unit/ngsiv2/examples/contextRequests/createExplicitAttrsDevice.json')
+            //     )
+            //     .reply(204);
```
Comment (fgalan) [inline test/unit/ngsiv2/provisioning/device-provisioning-api_test.js:346]: Lefover comment?

Comment (fgalan) [reply test/unit/ngsiv2/provisioning/device-provisioning-api_test.js:346]: Review it in general. It seems there is more cases.

File: CHANGES_NEXT_RELEASE [hunk 3]
```diff
@@ -1,3 +1,4 @@
+- Add: do not create initial entity when a new device is provisioned and appendMode is false
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE:1]:
```suggestion
- Add: do not create initial entity when a new device is provisioned and appendMode is false or NGSI-LD is used
```

File: test/unit/ngsiv2/provisioning/device-update-registration_test.js [hunk 4]
```diff
@@ -205,13 +205,13 @@ describe('NGSI-v2 - IoT Agent Device Update Registration', function () {
...
-                done();
-            });
-        });
+        // it('should register as ContextProvider of its lazy attributes', function (done) {
+        //     iotAgentLib.updateRegister(deviceUpdated, false, function (error) {
+        //         should.not.exist(error);
+        //         contextBrokerMock.done();
+        //         done();
+        //     });
+        // });
```
Comment (fgalan) [inline test/unit/ngsiv2/provisioning/device-update-registration_test.js:214]:
```suggestion
```

File: test/unit/ngsiv2/provisioning/device-update-registration_test.js [hunk 5]
```diff
@@ -260,13 +260,13 @@ describe('NGSI-v2 - IoT Agent Device Update Registration', function () {
...
-                done();
-            });
-        });
+        // it('should register as ContextProvider of its commands and create the additional attributes', function (done) {
+        //     iotAgentLib.updateRegister(deviceCommandUpdated, false, function (error) {
+        //         should.not.exist(error);
+        //         contextBrokerMock.done();
+        //         done();
+        //     });
+        // });
```
Comment (fgalan) [inline test/unit/ngsiv2/provisioning/device-update-registration_test.js:269]:
```suggestion
```

File: test/unit/plugins/capture-provision-inPlugins_test.js [hunk 6]
```diff
@@ -124,18 +124,18 @@ describe('NGSI-v2 - Data Mapping Plugins: device provision', function () {
...
+        //         callback(null, device);
+        //     }
+
+        //     iotAgentLib.addDeviceProvisionMiddleware(testMiddleware);
+
+        //     request(options, function (error, response, body) {
+        //         contextBrokerMock.done();
+        //         done();
+        //     });
+        // });
```
Comment (fgalan) [inline test/unit/plugins/capture-provision-inPlugins_test.js:138]:
```suggestion
```

**Review by fgalan — APPROVED** [review]
LGTM

**Pull request merged**